src/Controllers/SettingsController.php

```diff
@@ -21,11 +34,54 @@ class SettingsController extends BaseController
      * @param Response $response
      */
     public function __construct(
+        Authenticator $auth,
         Config $config,
+        LoggerInterface $log,
+        Redirector $redirector,
         Response $response
     ) {
-        $this->config = $config;
-        $this->response = $response;
+            $this->auth = $auth;
+            $this->config = $config;
+            $this->log = $log;
+            $this->redirect = $redirector;
+            $this->response = $response;
+    }
+
+    /**
+     * @return Response
+     */
+    public function password(): Response
+    {
+        return $this->response->withView(
+            'pages/settings/password.twig',
+            $this->getNotifications()
+        );
+    }
+
+    /**
+     * @return Response
+     */
+    public function savePassword(Request $request): Response
+    {
+        $user = $this->auth->user();
+
+        if (
+            !$request->has('password')
+            || !$this->auth->verifyPassword($user, $request->postData('password'))
+        ) {
+            $this->addNotification('-> not OK. Please try again.', 'errors');
```

Comment:
These should definitely be translation strings (and esp. this message should be reworded as its not that descriptive)